src/tools/simeng/gdb/GDBStub.cc

```diff
@@ -0,0 +1,375 @@
+#include "simeng/GDBStub.hh"
+
+namespace simeng {
+
+GDBStub::GDBStub(simeng::Core& core, simeng::MemoryInterface& dataMemory)
+    : core_(core), dataMemory_(dataMemory) {}
+
+int GDBStub::run() {
+  // verbose_ = true; // uncomment this if you'd like the GDBStub to run
+  // verbosely
+
+  int connection = openSocket(2424);  // change this if the port is blocked
+
+  std::cout << "[SimEng:GDBStub] Connection to GDB client established, "
+               "debugging in progress"
+            << std::endl
+            << std::endl;
+
+  char buffer[10000];
+
+  while (true) {
+    read(connection, buffer, 10000);
+
+    // First parentheses = package, second = checksum
+    std::regex packet_regex("^\\+*\\$([^#]*)#([0-9a-f]{2})");
```

Comment:
Would improve code readability if there was a brief explanation of the RegEx used

Reply:
I have included some explanations for this and other regexes (and added slightly tighter logic) in [this commit](https://github.com/UoB-HPC/SimEng/pull/244/commits/ec4a185f36b67eeffa59ae11d73c2830529a3c16)